### DISCUSSION TITLE
How exactly am I supposed to use Scene preview?

### DISCUSSION THREAD

**Discussion by SamDevelopsCode** [godotengine/godot-vscode-plugin#689] [General]
Just trying to figure out intended behavior. 
1. If I open/select a script that is on the root node of a PackedScene, should it show that scene's tree structure in the Scene Preview? Or do I have to click on the .tscn?
2. How should it work with inherited scenes?
3. If I open/select a subscene ( a scene that inherits another scene) should it show me the parent scene's tree? Or the subscene's tree which I currently have the script opened?

**Comment by DaelonSuzuka** [comment]
You're way overthinking it. All it does is parse a scene file when you open one. If you open a script, it tries to find a scene file that matches and it parses that. It doesn't follow any links or know anything about inheritance or subscenes or packed scenes.

**Comment by SamDevelopsCode** [comment]
"If you open a script, it tries to find a scene file that matches and it parses that."

Then it doesn't seem to work very much at all with how you have described. Below is a video. 


1. If I click on a script the scene preview doesn't seem to change at all. 
2. If I click on a scene file it updates the scene preview but if its an inherited scene it only shows packedscenes from the parent scene and nodes that are specific to this child scene.

I have anyFolder in settings selected. Version 2.1 installed
*Soldier and Charger inherit from Humanoid*

https://github.com/user-attachments/assets/a5ee82b7-4e7a-4ec5-86f7-0521bf009920

I am just trying to figure out the intended behavior because this seems like a bug based on your statements.

**Reply by DaelonSuzuka** [reply]
It matches the filename, and by default it only looks for a matching scene in the same directory as the script you open. 

You can change this setting to loosen that restriction:

![image](https://github.com/user-attachments/assets/c28d00c7-d519-4b1a-af1f-b30f4c19510b)

**Reply by SamDevelopsCode** [reply]
Like I said in the previous message, I already have anyFolder selected and it still doesn't open the correct scene in the preview by clicking on script that has the same name as the scene.

Is there a capitalization issue perhaps when parsing? I name my files using snake_case and my node name's in Pascal.

**Reply by DaelonSuzuka** [reply]
> Like I said in the previous message

I must admit, I was drawn directly to the video.

> Is there a capitalization issue perhaps when parsing? I name my files using snake_case and my node name's in Pascal.

No, the contents of the scene don't influence the matching behavior, only the filenames. 

> I am just trying to figure out the intended behavior because this seems like a bug based on your statements.

Yeah, this does look like wierd behavior. I'll look into this on my end.
